shard/src/ext/purge_command.py

```diff
@@ -1,41 +1,192 @@
+                await ctx.send(f"Deleted {len(deleted)} message(s)")
+    except discord.Forbidden:
+        await ctx.send("Architus does not have permission to remove messages in this server")
+    except discord.HTTPException:
+        await ctx.send("Failed to reach discord serverrs. Try again in a bit")
+    except Exception:
+        await ctx.send("Something went wrong. :(")
+
+
+@purge.command()
+async def time(ctx, time_window):
+    """
+    Purge a channel of messages sent in the past X amount of time
+
+    Usage: {prefix}purge time {XXm|XXs}
+
+    Deletes all messages in the channel in the past number of
+    minutes or seconds.
+
+    NOTE: If the original message is deleted before the bot deletes it, all messages
+    from the target message's author in the channel will be deleted depending on the
+    timing of when the original message was deleted.
+    """
+    # Generally considered bad practice to use the utcnow datetime function.
+    # However, the discord api specifically requires a timezone niave datetime
+    # that represents the relevant time in the UTC timezone. This is exactly
+    # what utcnow gives us so that's what we'll use.
+    # See: https://discordpy.readthedocs.io/en/latest/api.html?highlight=channel#discord.TextChannel.history
+    now = datetime.utcnow()
+    time_param = TIME_REGEX.match(time_window.strip())
+    if time_param is None:
+        await ctx.send("Time value formatted improperly")
+        return
+
+    if time_param[2] == 'm':
+        mins = int(time_param[1])
+        diff = timedelta(minutes=mins)
+    elif time_param[2] == 's':
+        secs = int(time_param[1])
+        diff = timedelta(seconds=secs)
+
+    earliest = now - diff
+
+    settings = ctx.bot.settings[ctx.guild]
+    if ctx.author.id not in settings.admins_ids:
+        await ctx.send("You do not have permissions to purge messaages")
+        return
+
+    try:
+        messages_to_delete = await ctx.channel.history(limit=500, after=earliest).flatten()
+    except discord.Forbidden:
+        await ctx.send("Architus does not have permission to read message history")
+        return
+    except discord.HTTPException:
+        await ctx.send("Failed to reach discord servers. Try again in a bit")
+        return
+    except Exception:
+        await ctx.send("Something went wrong. :(")
+        return
+
+    try:
+        if len(messages_to_delete) <= 100:
+            async with ctx.channel.typing():
+                await ctx.channel.delete_messages(messages_to_delete)
+                await ctx.send(f"Deleted {len(messages_to_delete)} message(s)")
         else:
-            await ctx.send(f'lul {ctx.author.mention}')
+            ids = list(map(lambda m: m.id, messages_to_delete))
+            async with ctx.channel.typing():
+                deleted = await ctx.channel.purge(limit=len(messages_to_delete), check=lambda m: m.id in ids,
+                                                  bulk=False)
```

Comment:
what does `bulk=False` do?

Reply:
I'm not entirely sure. But reading over the documentation again it might be better to switch it to true. You can see if you can understand it https://discordpy.readthedocs.io/en/latest/api.html?highlight=textchannel#discord.TextChannel.purge